### PR TITLE
Adding tests by neubaufx

### PR DESCRIPTION
**Pull request by Logende**

### CODE REVIEW

File: tests/integration/test_diffusion2d.py [hunk 1]
```diff
@@ -11,9 +12,42 @@ def test_initialize_physical_parameters():
...
+    h = 4.0
+    T_cold = 100
+    T_hot = 500
+
+    solver.initialize_domain(w, h, dx, dy)
+    solver.initialize_physical_parameters(d, T_cold, T_hot)
+
+    actual_result = solver.set_initial_condition()
+
+    expected_result = [[100.0, 100.0]]
```
Comment (IshaanDesai) [inline tests/integration/test_diffusion2d.py:48]: I am a bit confused by this, how is it the case that you do not need to set 500 in the expected result?

File: tests/unit/test_diffusion2d_functions.py [hunk 2]
```diff
@@ -1,26 +1,79 @@
...
+        solver = self.solver
+        solver.T_cold = 100
+        solver.T_hot = 500
+        solver.nx = 1
+        solver.ny = 4
+        solver.dx = 1.0
+        solver.dy = 2.0
+
+        # Act
+        actual_result = solver.set_initial_condition()
```
Comment (IshaanDesai) [inline tests/unit/test_diffusion2d_functions.py:72]: Same confusion as above.

File: tests/unit/test_diffusion2d_functions.py [hunk 3]
```diff
@@ -1,26 +1,79 @@
...
+
+        # Assert
+        # based on my extensive physics knowledge, I have computed that with the given input, the result must be this
+        expected_result = [[100.0, 100.0, 100.0, 100.0]]
```
Comment (IshaanDesai) [inline tests/unit/test_diffusion2d_functions.py:76]: Same confusion as above.

**Review by IshaanDesai — APPROVED** [review]

File: tests/unit/test_diffusion2d_functions.py (hunk 3 above)
Comment (Logende) [reply tests/unit/test_diffusion2d_functions.py:76]: In full disclosure: at the time of writing those tests I was not interested in trying to understand the underlying physics function. I took the cheating shortcut of putting in some input, seeing what the result is and then just using that as a test. Of course, in a real life scenario that’s not how someone would test if their function was implemented properly. I jokingly hint at that in the (ironic) comment:
“based on my extensive physics knowledge, I have computed that with the given input, the result must be this”

**Pull request closed without merging**